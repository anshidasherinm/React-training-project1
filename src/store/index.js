import { configureStore } from "@reduxjs/toolkit";
import todoSlice from "store/todo/todo-slice";
import labelSlice from "store/label/label-slice";

const store = configureStore({
  reducer: {
    todo: todoSlice.reducer,
    label: labelSlice.reducer,
  },
});

export default store;
