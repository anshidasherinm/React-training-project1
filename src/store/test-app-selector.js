const state = {
  todo: {
    items: [
      {
        checked: false,
        id: 714,
        key: "-NOqX_ZRWmYBbJS0hkJh",
        labelKey: "-NOqXGK_kreZB5AW_PFG",
        title: "study for exam",
      },
    ],
    allItems: [],
    addingTodoItem: false,
    addingTodoItemError: false,
    loadingList: false,
    loadingListError: false,
    updatingTodo: false,
    updatingTodoError: false,
    deletingTodoItem: false,
    deletingTodoItemError: false,
  },
  label: {
    labels: [{ id: 1463, key: "-NOqXGK_kreZB5AW_PFG", name: "School" }],
    addingLabel: false,
    addingLabelError: false,
    loadingLabels: false,
    loadingLabelError: false,
    updatingLabel: false,
    updatingLabelError: false,
    deletingLabel: false,
    deletingLabelError: false,
  },
};

export const testUseAppSelector = (f) => f(state);
