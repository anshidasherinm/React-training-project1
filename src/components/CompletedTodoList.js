import React from "react";
import TodoItem from "./TodoItem";
import classes from "./CompletedTodoList.module.css";
const CompletedTodoList = (props) => {
  const addCheckHandler = (id) => {
    props.addCheck(id);
  };
  const list = props.list;
  const completedTodoList = list.map((todo) => (
    <TodoItem
      id={todo.id}
      title={todo.title}
      key={todo.id}
      check={todo.checked}
      addCheck={addCheckHandler}
    />
  ));
  const emptylist = <div className={classes.emptylist}>No completed todos</div>;
  const content = list.length > 0 ? completedTodoList : emptylist;
  return <div className={props.className}>{content}</div>;
};

export default CompletedTodoList;
