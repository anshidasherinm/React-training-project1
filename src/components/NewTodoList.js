import React from "react";
import { useSelector } from "react-redux";
import classes from "./NewTodoList.module.css";
import TodoItem from "./TodoItem";

const NewTodoList = (props) => {
  const todoItems = useSelector((state) =>
    state.todo.items.filter((todo) => todo.checked === false)
  );
  // const todos = useSelector((state) => state.todo.items);
  // console.log(todos);
  console.log(todoItems);

  if (todoItems.length === 0) {
    return <div className={classes.emptylist}>Add a todo</div>;
  }

  return (
    <div className={props.className}>
      {todoItems.map((todo) => {
        return (
          <TodoItem
            id={todo.key}
            title={todo.title}
            key={todo.key}
            check={todo.checked}
          />
        );
      })}
    </div>
  );
};

export default NewTodoList;
