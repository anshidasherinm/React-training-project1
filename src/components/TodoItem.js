import React, { useMemo } from "react";
import { useDispatch, useSelector } from "react-redux";
import classes from "./TodoItem.module.css";
import { updateTodo, deleteTodo } from "../store/todo/todo-thunk";
import todoSlice, { setKey } from "../store/todo/todo-slice";

const TodoItem = (props) => {
  const dispatch = useDispatch();
  const todoItems = useSelector((state) => state.todo.items);
  const todoItemsAll = useSelector((state) => state.todo.allItems);

  const checkBoxClickHandler = (key, e) => {
    if (e.target.checked) {
      const item = todoItems.find((e) => e.key === key);
      dispatch(updateTodo({ ...item, checked: true }));
    } else {
      const item = todoItems.find((e) => e.key === key);

      dispatch(updateTodo({ ...item, checked: false }));
    }
  };

  const removeTodoHandler = (key) => {
    const item = todoItems.find((e) => e.key === key);
    dispatch(deleteTodo({ ...item, checked: true }));
  };

  return (
    <div className={classes.todo}>
      <input
        type="checkbox"
        checked={props.check}
        id={props.id}
        name={props.title}
        onChange={(e) => checkBoxClickHandler(props.id, e)}
      />
      <div className={classes.title}>{props.title}</div>
      {!props.check ? (
        <i
          className={`fa fa-remove ${classes.remove}`}
          onClick={(e) => removeTodoHandler(props.id)}
        />
      ) : null}
    </div>
  );
};

export default React.memo(TodoItem);
