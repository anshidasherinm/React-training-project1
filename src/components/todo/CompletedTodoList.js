import React from "react";
import { useAppSelector } from "store/redux-hooks";
import TodoItem from "components/todo/TodoItem";
import { Card, List } from "@mui/material";

const CompletedTodoList = (props) => {
  const styles = {
    card_empty: {
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
      alignItems: "center",
      overflowY: "auto",
      width: "100%",
      color: "white",
      backgroundColor: "#383838",
    },
    card_notempty: {
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
      alignItems: "center",
      width: "100%",
      backgroundColor: "rgb(42, 4, 77)",
    },
    list: {
      width: "100%",
      bgcolor: "background.paper",
      padding: "0px",
      backgroundColor: "#383838",
      overflowY: "auto",
      display: "flex",
      flexDirection: "column",
      height: "100%",
      alignItems: "center",
    },
  };

  const todoItems = useAppSelector((state) =>
    state.todo.items.filter((todo) => todo.checked === true)
  );

  if (todoItems.length === 0) {
    return <Card style={styles.card_empty}>No completed todos</Card>;
  }

  return (
    <Card style={styles.card_notempty}>
      <List sx={styles.list} data-testid="todo-list">
        {todoItems.map((todo, index) => {
          const labelId = `checkbox-list-label-${todo.id}`;

          return (
            <TodoItem
              data-testid={`item-${index}`}
              id={todo.key}
              title={todo.title}
              key={todo.key}
              check={todo.checked}
              labelId={labelId}
            />
          );
        })}
      </List>
    </Card>
  );
};

export default CompletedTodoList;
