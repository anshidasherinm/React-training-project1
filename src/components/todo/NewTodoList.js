import React from "react";
import { Card, List } from "@mui/material";
import TodoItem from "components/todo/TodoItem";
import { useAppSelector } from "store/redux-hooks";

const NewTodoList = (props) => {
  const todoItems = useAppSelector((state) =>
    state.todo.items.filter((todo) => todo.checked === false)
  );

  const styles = {
    card_empty: {
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
      alignItems: "center",
      overflowY: "auto",
      width: "100%",
      color: "white",
      backgroundColor: "#383838",
    },
    card_notempty: {
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
      alignItems: "center",
      width: "100%",
      backgroundColor: "#383838",
    },
    list: {
      padding: "0px",
      width: "100%",
      bgcolor: "background.paper",
      overflowY: "auto",
      display: "flex",
      flexDirection: "column",
      height: "100%",
      alignItems: "center",
      backgroundColor: "#383838",
    },
  };

  if (todoItems.length === 0) {
    return <Card style={styles.card_empty}>Add a todo</Card>;
  }

  return (
    <Card style={styles.card_notempty}>
      <List sx={styles.list}>
        {todoItems.map((todo) => {
          const labelId = `checkbox-list-label-${todo.id}`;

          return (
            <TodoItem
              id={todo.key}
              title={todo.title}
              key={todo.key}
              check={todo.checked}
              lalabelId={labelId}
            />
          );
        })}
      </List>
    </Card>
  );
};

export default NewTodoList;
