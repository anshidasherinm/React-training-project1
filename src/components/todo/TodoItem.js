import React from "react";
import { updateTodo, deleteTodo } from "store/todo/todo-thunk";
import { useAppDispatch, useAppSelector } from "store/redux-hooks";
import { makeStyles } from "@material-ui/core/styles";
import { Checkbox, ListItemIcon, ListItemText, ListItem } from "@mui/material";

const useStyles = makeStyles({
  remove: {
    color: "red",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    width: "1rem",
    fontSize: "larger",
    marginRight: "1rem",
    "&:hover": {
      color: "rgb(243, 65, 65)",
    },

    checkBoxStyle: {
      color: "white",
      // "&.Mui-checked": classes.muiChecked
      "&.Mui-checked": { color: "orange" },
    },
  },

  muiChecked: {
    color: "orange",
  },
});

const TodoItem = (props) => {
  // const dispatch = useDispatch();
  // const todoItems = useSelector((state) => state.todo.items);
  const dispatch = useAppDispatch();
  const todoItems = useAppSelector((state) => state.todo.items);

  const classes = useStyles();

  const checkBoxClickHandler = (key, e) => {
    if (e.target.checked) {
      const item = todoItems.find((e) => e.key === key);
      dispatch(updateTodo({ ...item, checked: true }));
    } else {
      const item = todoItems.find((e) => e.key === key);

      dispatch(updateTodo({ ...item, checked: false }));
    }
  };

  const removeTodoHandler = (key) => {
    const item = todoItems.find((e) => e.key === key); //expression
    dispatch(deleteTodo({ ...item, checked: true })); //statement
  };

  const styles = {
    //above function
    listItem: {
      paddingLeft: "1rem",
      width: "100%",
      boxSizing: "border-box",
      backgroundColor: "#383838",
      minHeight: "5rem",
      borderBottom: "3px solid #282828",
    },

    checkbox: {
      color: "white",
      // "&.Mui-checked": classes.muiChecked
      "&.Mui-checked": { color: "orange" },
    },

    listItemText: {
      overflowWrap: "normal",
      "& .MuiTypography-root": {
        color: "orange",
        fontSize: "1.1rem",
        fontWeight: "normal",
      },
      marginRight: "2rem",
    },
    listItemIcon1: { minWidth: "40px" },
    listItemIcon2: { minWidth: "40px", marginLeft: "1rem" },
  };

  return (
    <ListItem
      sx={styles.listItem}
      key={props.id}
      disablePadding
      // data-testid={"todo-listItem"}
    >
      <ListItemIcon sx={styles.listItemIcon1}>
        <Checkbox
          // sx={classes.checkBoxStyle}
          // className={classes.checkBoxStyle}
          sx={styles.checkbox}
          edge="start"
          checked={props.check}
          onChange={(e) => checkBoxClickHandler(props.id, e)}
          inputProps={{ "aria-labelledby": props.labelId }}
          data-testid={"todo-checkbox"}
        />
      </ListItemIcon>

      <ListItemText
        id={props.labelId}
        primary={`${props.title}`}
        sx={styles.listItemText}
        data-testid={"todo-listItemtext"}
      />

      <ListItemIcon sx={styles.listItemIcon2} data-testid={"todo-removeicon"}>
        {!props.check ? (
          <i
            className={`fa fa-remove ${classes.remove}`}
            onClick={(e) => removeTodoHandler(props.id)}
          />
        ) : null}
      </ListItemIcon>
    </ListItem>
  );
};

export default TodoItem;
