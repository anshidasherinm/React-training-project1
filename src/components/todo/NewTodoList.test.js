import { render, screen } from "@testing-library/react";
import CompletedTodoList from "./CompletedTodoList";
import { useAppDispatch, useAppSelector } from "store/redux-hooks";
import { testUseAppSelector } from "store/test-app-selector";
import "@testing-library/jest-dom";
import NewTodoList from "./NewTodoList";
import { shallow } from "enzyme";
import TodoItem from "./TodoItem";

jest.mock("store/redux-hooks");
describe("NewRodoList", () => {
  beforeEach(() => {
    useAppSelector.mockImplementation(testUseAppSelector);
    // const dispatch = jest.fn();
    useAppDispatch.mockReturnValue(jest.fn());
  });
  //   it("renders Add a todo ", () => {
  //     //Arrange
  //     render(<NewTodoList />);

  //     //Act
  //     //...nothing

  //     //Assert
  //     const emptyList = screen.getByText("Add a todo");
  //     expect(emptyList).toBeInTheDocument();
  //   });
  // });
  it("We show a todo  in NewTodoList", async () => {
    render(<NewTodoList />);

    const enzymeWrapper = shallow(
      <TodoItem
        id={714}
        key="-NOqX_ZRWmYBbJS0hkJh"
        title="study for exam"
        labelKey="-NOqXGK_kreZB5AW_PFG"
        checked={false}
      />
    );
    expect(enzymeWrapper.text()).toMatchSnapshot();
  });
});
