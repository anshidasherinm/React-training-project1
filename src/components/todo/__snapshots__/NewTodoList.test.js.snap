// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewRodoList We show a todo  in NewTodoList 1`] = `""`;
